src/core/overlay/overlay.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  DynamicComponentLoader,
+  AppViewManager,
+  OpaqueToken,
+  Inject,
+  Injectable} from 'angular2/core';
+import {CONST_EXPR} from 'angular2/src/facade/lang';
+import {OverlayState} from './overlay-state';
+import {DomPortalHost} from '../portal/dom-portal-host';
+import {OverlayRef} from './overlay-ref';
+import {DOM} from '../platform/dom/dom_adapter';
+
+// Re-export OverlayState and OverlayRef so they can be imported directly from here.
+export {OverlayState} from './overlay-state';
+export {OverlayRef} from './overlay-ref';
+
+/** Token used to inject the DOM element that serves as the overlay container. */
+export const OVERLAY_CONTAINER_TOKEN = CONST_EXPR(new OpaqueToken('overlayContainer'));
+
+/** Next overlay unique ID. */
+let nextUniqueId = 0;
+
+/** The default state for newly created overlays. */
+let defaultState = new OverlayState();
+
+
+/**
+ * Service to create Overlays. Overlay are dynamically added pieces of floating UI, meant to be
+ * used as low-level a building building block for other components. Dialogs, tooltips, menus,
+ * selects, etc. can all be built using overlays. The service should primarily be used by authors
+ * of re-usable components rather than developers building end-user applications.
+ *
+ * An overlay *is* a PortalHost, so any kind of Portal can be loaded into one.
+ */
+ @Injectable()
+export class Overlay {
+  constructor(
+      @Inject(OVERLAY_CONTAINER_TOKEN) private _overlayContainerElement: Element,
+      private _dynamicComponentLoader: DynamicComponentLoader,
+      private _appViewManager: AppViewManager) {
+  }
+
+  /**
+   * Creates an overlay.
+   * @param state State to apply to the overlay.
+   * @returns A reference to the created overlay.
+   */
+  create(state: OverlayState = defaultState): Promise<OverlayRef> {
```

Comment:
I'm not sure default arguments work with TS2Dart yet. You may have to write this by checking if `isPresent`